src/braket/circuits/noise.py

```diff
@@ -331,85 +429,138 @@ def __init__(
         Raises:
             ValueError: If the `qubit_count` is less than 1, `ascii_symbols` are `None`, or
                 `ascii_symbols` length != `qubit_count`, `probX` or `probY` or `probZ`
-                is not `float`, `probX` or `probY` or `probZ` > 1.0, or
+                is not `float` or FreeParameterExpression, `probX` or `probY` or `probZ` > 1.0, or
                 `probX` or `probY` or `probZ` < 0.0, or `probX`+`probY`+`probZ` > 1
         """
         super().__init__(qubit_count=qubit_count, ascii_symbols=ascii_symbols)
 
-        if not isinstance(probX, float):
-            raise TypeError("probX must be float type")
-        if not (probX <= 1.0 and probX >= 0.0):
-            raise ValueError("probX must be a real number in the interval [0,1]")
-        if not isinstance(probY, float):
-            raise TypeError("probY must be float type")
-        if not (probY <= 1.0 and probY >= 0.0):
-            raise ValueError("probY must be a real number in the interval [0,1]")
-        if not isinstance(probZ, float):
-            raise TypeError("probZ must be float type")
-        if not (probZ <= 1.0 and probZ >= 0.0):
-            raise ValueError("probZ must be a real number in the interval [0,1]")
-        if probX + probY + probZ > 1:
+        total = 0
+        total += PauliNoise._validate_param("probX", probX)
+        total += PauliNoise._validate_param("probY", probY)
+        total += PauliNoise._validate_param("probZ", probZ)
+        if total > 1:
             raise ValueError("the sum of probX, probY, probZ cannot be larger than 1")
-
-        self._probX = probX
-        self._probY = probY
-        self._probZ = probZ
+        self._parameters = [probX, probY, probZ]
+
+    @staticmethod
+    def _validate_param(param_name, param: Union[FreeParameterExpression, float]):
```

Comment:
nit: docstrings

Reply:
Happy to add them, but just FYI, docstrings are only required on public methods; private docstrings will not be built into our Read the Docs documentation.